src/impl/apis/data_api_impl.py

```diff
@@ -400,3 +400,28 @@ async def run_query(
     )
 
     return new_result
+
+
+async def run_csv_query(
+    query_parameters: QueryParameters, pool: Pool, token_model: TokenModel
+) -> QueryResult:
+    result = await run_query(query_parameters, pool, token_model)
+    csv_rows = []
+    for date, data in result.data_by_date.items():
```

Comment:
Noting for posterity that this may cause us issues because we'll block the whole server while creating the csv. Ideally what we'd do here is stream the csv back async instead (appreciate that there's not scope to do that at this time).